flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -289,6 +294,10 @@ public Task(
 			ConfigConstants.TASK_CANCELLATION_INTERVAL_MILLIS,
 			ConfigConstants.DEFAULT_TASK_CANCELLATION_INTERVAL_MILLIS);
 
+		this.taskCancellationTimeout = jobConfiguration.getLong(
```

Comment:
I think the `jobConfiguration` is the wrong configuration here. It should probably be the TaskManager's configuration in the `taskManagerConfig`. Same probably holds for the cancellation interval above.

Seems neither were ever tested?